### PR TITLE
feat(RDS): Additional RDS checks

### PR DESCRIPTION
**Pull request by sansns**
### Context

1. Add default admin user check.
2. Backtrack is enabled on Aurora MySQL Clusters
3. IAM Authentication is enabled for DB Instances/Clusters
4. Snapshots are encrypted

### Description

1. 
```
All RDS DB instances/clusters use ```admin``` as a default username other than PostgreSQL based instances/clusters which use ```postgres``` as the default user. Added checks to find both ```admin``` and ```postgres``` being the default admin username. 

Trend Cloud Conformity and AWS RDS Controls only check for ```admin```, but I also added ```postgres``` as that is the default PostgreSQL username to the check are more robust.

https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/rds-master-username.html#
https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-25
```
2. 
```
Ensure that the Backtrack feature is enabled for your Amazon Aurora (with MySQL compatibility) database clusters in order to backtrack your clusters to a specific time, without using backups. Backtrack is an Amazon RDS feature that allows you to specify the amount of time that an Aurora MySQL database cluster needs to retain change records, in order to have a fast way to recover from user errors, such as dropping the wrong table or deleting the wrong row by moving your MySQL database to a prior point in time without the need to restore from a recent backup.

Trend Cloud Conformity has it as a low priority
AWS RDS Controls has it as medium priority

DB Clusters
https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/backtrack.html#
https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-14
https://docs.aws.amazon.com/config/latest/developerguide/aurora-mysql-backtracking-enabled.html
```
3.
 ```
With this feature enabled, you don't have to use a password when you connect to your MySQL/PostgreSQL database, instead you can use an authentication token. An authentication token is a unique string of characters with a lifetime of 15 minutes that Amazon RDS generates on your request. IAM Database Authentication removes the need of storing user credentials within the database configuration, because authentication is managed externally using Amazon IAM.

DB Instances
https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/iam-database-authentication.html#
https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-10

DB Clusters
https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-12
```
4.
```
Ensure that your manual Amazon RDS database snapshots are encrypted in order to achieve compliance for data-at-rest encryption within your organization. The Amazon RDS snapshot encryption and decryption process is handled transparently and does not require any additional action from you or your application. 

https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/snapshot-encrypted.html#
https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-4
```

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by snaow** [comment]
Thanks @sansns! We will review and merge/get back to you soon

File: prowler/providers/aws/services/rds/rds_cluster_backtrack_enabled/rds_cluster_backtrack_enabled.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "Risk": "Once the Backtrack feature is enabled, Amazon RDS can quickly \"rewind\" your Aurora MySQL database cluster to a point in time that you specify. In contrast to the backup and restore method, with Backtrack you can easily undo a destructive action, such as a DELETE query without a WHERE clause, with minimal downtime, you can rewind your Aurora cluster in just few minutes, and you can repeatedly backtrack a database cluster back and forth in time to help determine when a particular data change occurred.",
+  "RelatedUrl": "https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-14",
+  "Remediation": {
+    "Code": {
+      "CLI": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/backtrack.html#",
+      "NativeIaC": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/backtrack.html#",
+      "Other": "",
+      "Terraform": "https://docs.prowler.com/checks/aws/general-policies/ensure-that-rds-instances-have-backup-policy#terraform"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_cluster_backtrack_enabled/rds_cluster_backtrack_enabled.metadata.json]:
```suggestion
      "Terraform": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/backtrack.html#"
```

File: prowler/providers/aws/services/rds/rds_cluster_backtrack_enabled/rds_cluster_backtrack_enabled.metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "medium",
+  "ResourceType": "AwsRdsDbClsuter",
+  "Description": "Ensure that the Backtrack feature is enabled for your Amazon Aurora (with MySQL compatibility) database clusters in order to backtrack your clusters to a specific time, without using backups. Backtrack is an Amazon RDS feature that allows you to specify the amount of time that an Aurora MySQL database cluster needs to retain change records, in order to have a fast way to recover from user errors, such as dropping the wrong table or deleting the wrong row by moving your MySQL database to a prior point in time without the need to restore from a recent backup.",
+  "Risk": "Once the Backtrack feature is enabled, Amazon RDS can quickly \"rewind\" your Aurora MySQL database cluster to a point in time that you specify. In contrast to the backup and restore method, with Backtrack you can easily undo a destructive action, such as a DELETE query without a WHERE clause, with minimal downtime, you can rewind your Aurora cluster in just few minutes, and you can repeatedly backtrack a database cluster back and forth in time to help determine when a particular data change occurred.",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_cluster_backtrack_enabled/rds_cluster_backtrack_enabled.metadata.json]:
```suggestion
  "Risk": "Once the Backtrack feature is enabled, Amazon RDS can quickly 'rewind' your Aurora MySQL database cluster to a point in time that you specify. In contrast to the backup and restore method, with Backtrack you can easily undo a destructive action, such as a DELETE query without a WHERE clause, with minimal downtime, you can rewind your Aurora cluster in just few minutes, and you can repeatedly backtrack a database cluster back and forth in time to help determine when a particular data change occurred.",
```

File: prowler/providers/aws/services/rds/rds_instance_default_admin/rds_instance_default_admin.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            report.region = db_instance.region
+            report.resource_id = db_instance.id
+            report.resource_arn = db_instance.arn
+            report.resource_tags = db_instance.tags
+            report.status = "FAIL"
+            report.status_extended = (
+                f"RDS Instance {db_instance.id} is using the default master username."
+            )
+
+            # Check only RDS DB instances that support parameter group encryption
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_default_admin/rds_instance_default_admin.py]:
```suggestion
            # Check only RDS DB instances that are not clustered
```

File: prowler/providers/aws/services/rds/rds_instance_default_admin/rds_instance_default_admin.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        for db_cluster in rds_client.db_clusters:
+            report = Check_Report_AWS(self.metadata())
+            report.region = rds_client.db_clusters[db_cluster].region
+            report.resource_id = rds_client.db_clusters[db_cluster].id
+            report.resource_arn = db_cluster
+            report.resource_tags = rds_client.db_clusters[db_cluster].tags
+            report.status = "FAIL"
+            report.status_extended = f"RDS Cluster {rds_client.db_clusters[db_cluster].id} is using the default master username."
+            # Check RDS Clusters that support TLS encryption
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_default_admin/rds_instance_default_admin.py]:
```suggestion
```

File: prowler/providers/aws/services/rds/rds_service.py [hunk 5]
```diff
@@ -81,6 +81,10 @@ def __describe_db_instances__(self, regional_client):
                                         for item in instance["DBParameterGroups"]
                                     ],
                                     multi_az=instance["MultiAZ"],
+                                    username=instance["MasterUsername"],
+                                    iam_auth=instance[
+                                        "IAMDatabaseAuthenticationEnabled"
+                                    ],
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_service.py]:
```suggestion
                                    iam_auth=instance.get(
                                        "IAMDatabaseAuthenticationEnabled", False
                                    ),
```

File: prowler/providers/aws/services/rds/rds_service.py [hunk 6]
```diff
@@ -242,6 +248,10 @@ def __describe_db_clusters__(self, regional_client):
                                             "DBClusterParameterGroup"
                                         ],
                                         multi_az=cluster["MultiAZ"],
+                                        username=cluster["MasterUsername"],
+                                        iam_auth=cluster[
+                                            "IAMDatabaseAuthenticationEnabled"
+                                        ],
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_service.py]:
```suggestion
                                        iam_auth=cluster.get(
                                            "IAMDatabaseAuthenticationEnabled", False
                                        ),
```

File: prowler/providers/aws/services/rds/rds_service.py [hunk 7]
```diff
@@ -497,10 +510,13 @@ class DBCluster(BaseModel):
     public: bool
     encrypted: bool
     backup_retention_period: int = 0
+    backtrack: int = 0
     cloudwatch_logs: Optional[list]
     deletion_protection: bool
     auto_minor_version_upgrade: bool
     multi_az: bool
+    username: Optional[str]
+    iam_auth: Optional[bool]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_service.py]:
```suggestion
    username: str
    iam_auth: bool
```

File: prowler/providers/aws/services/rds/rds_service.py [hunk 8]
```diff
@@ -476,6 +487,8 @@ class DBInstance(BaseModel):
     auto_minor_version_upgrade: bool
     enhanced_monitoring_arn: Optional[str]
     multi_az: bool
+    username: Optional[str]
+    iam_auth: Optional[bool]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_service.py]:
```suggestion
    username: str
    iam_auth: bool
```

File: prowler/providers/aws/services/rds/rds_service.py [hunk 9]
```diff
@@ -497,10 +510,13 @@ class DBCluster(BaseModel):
     public: bool
     encrypted: bool
     backup_retention_period: int = 0
+    backtrack: int = 0
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_service.py]:
```suggestion
    backtrack: int
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]

**Review by MrCloudSec — APPROVED** [review]
These checks are awesome, thanks @sansns !

**Pull request merged**